domain/historical_stats.go

```diff
@@ -48,18 +49,26 @@ func (w *HistoricalStatsWorker) DumpHistoricalStats(tableID int64, statsHandle *
 	}
 	sctx := w.sctx
 	is := GetDomain(sctx).InfoSchema()
+	isPartition := false
+	var tblInfo *model.TableInfo
 	tbl, existed := is.TableByID(tableID)
 	if !existed {
-		return errors.Errorf("cannot get table by id %d", tableID)
+		tbl, db, p := is.FindTableByPartitionID(tableID)
+		if tbl != nil && db != nil && p != nil {
+			isPartition = true
+			tblInfo = tbl.Meta()
+		} else {
+			return errors.Errorf("cannot get table by id %d", tableID)
+		}
+	} else {
+		tblInfo = tbl.Meta()
 	}
-	tblInfo := tbl.Meta()
 	dbInfo, existed := is.SchemaByTable(tblInfo)
 	if !existed {
 		return errors.Errorf("cannot get DBInfo by TableID %d", tableID)
 	}
-	if _, err := statsHandle.RecordHistoricalStatsToStorage(dbInfo.Name.O, tblInfo); err != nil {
-		generateHistoricalStatsFailedCounter.Inc()
-		return errors.Errorf("record table %s.%s's historical stats failed", dbInfo.Name.O, tblInfo.Name.O)
+	if _, err := statsHandle.RecordHistoricalStatsToStorage(dbInfo.Name.O, tblInfo, tableID, isPartition); err != nil {
+		return errors.Errorf("record table %s.%s's historical stats failed, err:%v", dbInfo.Name.O, tblInfo.Name.O, err)
```

Comment:
We don't need `generateHistoricalStatsFailedCounter` here anymore?

Reply:
updated.